### ISSUE TITLE
version v1.1.0 use formatter property have issue

### ISSUE THREAD

**Issue by micro-sugar** [serilog-contrib/Serilog.Sinks.AmazonS3#13]
I can't not use Serilog.Sinks.AmazonS3 v1.1.0

code:
```
            string sysetmName = "sysetmName";
            string awsS3BucketName = "logforservice";
            string awsAccessKeyId = "OOO";
            string awsSecretAccessKey = "XXX";

            Log.Logger = new LoggerConfiguration()
                .MinimumLevel.Debug()
                .WriteTo.AmazonS3(
                    "log.log",
                    awsS3BucketName, 
                    Amazon.RegionEndpoint.APNortheast1,
                    awsAccessKeyId,
                    awsSecretAccessKey,
                    formatter : new Serilog.Formatting.Compact.CompactJsonFormatter(), 
                    bucketPath: $"{sysetmName}/{DateTime.Now.Year}/{DateTime.Now.Month}/{DateTime.Now.Day}/"
                    )
                .CreateLogger();
```

Exception:
```
System.ArgumentNullException
  HResult=0x80004003
  Message=Value cannot be null. 
  Source=Serilog
  StackTrace: 
   at Serilog.Formatting.Display.MessageTemplateTextFormatter..ctor(String outputTemplate, IFormatProvider formatProvider)
   at Serilog.Sinks.AmazonS3.AmazonS3Sink..ctor(AmazonS3Options amazonS3Options)
   at Serilog.LoggerConfigurationAmazonS3Extensions.AmazonS3(LoggerSinkConfiguration sinkConfiguration, String path, String bucketName, RegionEndpoint endpoint, String awsAccessKeyId, String awsSecretAccessKey, LogEventLevel restrictedToMinimumLevel, LoggingLevelSwitch levelSwitch, ITextFormatter formatter, RollingInterval rollingInterval, Encoding encoding, Action`1 failureCallback, String bucketPath, Nullable`1 batchSizeLimit, Nullable`1 batchingPeriod, Nullable`1 eagerlyEmitFirstEvent, Nullable`1 queueSizeLimit)
   at DotNetCoreSerilog.Program.Main(String[] args) in C:\Users\chase\Desktop\Serilog\DotNetCoreSerilog\DotNetCoreSerilog\Program.cs:line 29

```

**Comment by SeppPenner** [comment]
I will have a look at this soon, will take till tomorrow or even Sunday, though. I guess, I already have a clue where this comes from.

**Comment by SeppPenner** [comment]
I now can confirm that this is indeed a bug. It seems that this is a bug with my changes to using an options class internally. E.g. the sink now tries to always build a `ITextFormatter` by using the specified `OutputTemplate` and `IFormatProvider`. If you specify a formatter directly, both other options are `null`, of course. So, there is a simple null check missing. I will add this now and release a new package soon.

**Comment by SeppPenner** [comment]
This is fixed in version 1.1.1 which should be out in a few minutes.
